doc/error-handling.md

```diff
@@ -0,0 +1,324 @@
+Contents
+1. [Logging](#Logging)
+2. [Error Handling](#Error-Handling)
+3. [Usage](#Usage)
+4. [Open Points](#Open-Points)
+5. [Future Improvements](#Future-Improvements)
+
+
+# Logging
+Logging is performed by the iceoryx internal logger. The logger API implements a subset of the Autosar log and trace interface ara::log.
+The logger is used internally to record errors and general system runtime information. It can also be used by the user in application code for the same purpose.
+
+## Logger
+The logger is responsible for logging information about the state of the system to a configurable stream (includes files). 
+
+The logger is thread-safe and can hence be safely used from multiple threads concurrently. Currently the logger is synchronous.
+
+## Log Levels
+
+The following log levels are supported, ordered by criticality from lowest to highest.
+
+* VERBOSE - all available information is printed
```

Comment:
This is from the obsolete simplelogger. The current logger follows the [log4j](https://logging.apache.org/log4j/2.x/log4j-api/apidocs/org/apache/logging/log4j/Level.html) levels, except for `ALL`.

Reply:
Do we have the OFF level?

Reply:
I will correct this. Do we also have the OFF level?

Reply:
oh, my bad. Forget what I wrote

Reply:
We have no TRACE but I was still missing the OFF level.

Reply:
I'm currently checking the removal of simplelogger.